Kinvey.Core/Store/PushRequest.cs

```diff
@@ -15,180 +15,360 @@
+                var localEntity = Cache.FindByID(pwa.entityId);
 
-				NetworkRequest<T> request = Client.NetworkFactory.buildUpdateRequest<T>(pwa.collection, entity, pwa.entityId);
-				entity = await request.ExecuteAsync();
+                NetworkRequest<T> request = Client.NetworkFactory.buildUpdateRequest<T>(pwa.collection, localEntity, pwa.entityId);
+                entity = await request.ExecuteAsync();
 
                 Cache.UpdateCacheSave(entity, pwa.entityId);
 
                 result = SyncQueue.Remove(pwa);
 
-				if (result == 0)
-				{
-					offset++;
-				}
-			}
-			catch (KinveyException ke)
-			{
-				response.AddKinveyException(ke);
-				offset++;
-			}
-
-			return entity;
-		}
-
-		private async Task<T> HandlePushDELETE(PendingWriteAction pwa)
-		{
-			try
-			{
-				int result = 0;
-
-				NetworkRequest<KinveyDeleteResponse> request = Client.NetworkFactory.buildDeleteRequest<KinveyDeleteResponse>(pwa.collection, pwa.entityId);
-				KinveyDeleteResponse kdr = await request.ExecuteAsync();
-
-				if (kdr.count == 1)
-				{
-					result = SyncQueue.Remove(pwa);
-
-					if (result == 0)
-					{
-						offset++;
-					}
-				}
-			}
-			catch (KinveyException ke)
-			{
-				response.AddKinveyException(ke);
-				offset++;
-			}
-
-			return default(T);
-		}
-	}
+                if (result == 0)
+                {
+                    offset++;
+                }
+            }
+            catch (KinveyException ke)
+            {
+                kinveyException = ke;
+                offset++;
+            }
+
+            return new Tuple<T, KinveyException, int>(entity, kinveyException, offset);
+        }
+
+        private async Task<Tuple<T, KinveyException, int>> HandlePushDELETE(PendingWriteAction pwa)
+        {
+            var offset = 0;
+            KinveyException kinveyException = null;
+
+            try
+            {
+                int result = 0;
+
+                NetworkRequest<KinveyDeleteResponse> request = Client.NetworkFactory.buildDeleteRequest<KinveyDeleteResponse>(pwa.collection, pwa.entityId);
+                KinveyDeleteResponse kdr = await request.ExecuteAsync();
+
+                if (kdr.count == 1)
+                {
+                    result = SyncQueue.Remove(pwa);
+
+                    if (result == 0)
+                    {
+                        offset++;
+                    }
+                }
+            }
+            catch (KinveyException ke)
+            {
+                kinveyException = ke;
+                offset++;
+            }
+
+            return new Tuple<T, KinveyException, int>(default(T), kinveyException, offset);
+        }
+
+        #endregion Single actions pushes
+
+        #region Multiple actions pushes
+
+        private async Task<PushDataStoreResponse<T>> PushMultiPostActionsAsync()
+        {
+            var response = new PushDataStoreResponse<T>();
+            var limit = 10 * Constants.NUMBER_LIMIT_OF_ENTITIES;
```

Comment:
Isn't `Constants.NUMBER_LIMIT_OF_ENTITIES = 100`? if so, this will make the limit to `1,000` right? Where it should be `100`.

Reply:
@heyzooi 

You are right. The “Constants.NUMBER_LIMIT_OF_ENTITIES” is 100.
The variable “limit” will have 1000. It is correct too.
But the variable “limit” is responsible for receiving 1000 records from the local database.
10 is the maximum count of threads in one iteration.
It means that in one iteration we will fetch 1000 records from the local database and create 10 parallel threads for multi insert requests. Every request will have 100 records.
After this we will have the next iteration until we handle all needed records from the local database.

Reply:
oh I see, make sense! Thanks for the clarification